src/Numerics/LinearAlgebra/Matrix.Arithmetic.cs

```diff
@@ -1512,38 +1529,77 @@ public void PointwiseRemainder(Matrix<T> divisor, Matrix<T> result)
         }
 
         /// <summary>
-        /// Pointwise applies the exponent function to each value.
+        /// Pointwise applied the supplied unary operator to each value
         /// </summary>
-        public Matrix<T> PointwiseExp()
+        /// <param name="f"></param>
+        /// <returns></returns>
+        protected Matrix<T> PointwiseUnary(Action<Matrix<T>> f)
         {
             var result = Build.SameAs(this);
-            DoPointwiseExp(result);
+            f(result);
             return result;
         }
 
         /// <summary>
-        /// Pointwise applies the exponent function to each value.
+        ///  
         /// </summary>
-        /// <param name="result">The matrix to store the result.</param>
-        /// <exception cref="ArgumentException">If this matrix and <paramref name="result"/> are not the same size.</exception>
-        public void PointwiseExp(Matrix<T> result)
+        /// <param name="f"></param>
+        /// <returns></returns>
+        protected void PointwiseUnary(Action<Matrix<T>> f, Matrix<T> result)
```

Comment:
These are really because I saw the checks done in all the other pointwise functions before this pull request.  Since I am adding some 20 more of them, copy and pasting the checks is something I really want to avoid.  True, I did not try to refactor the existing ones though.  Yeah, please have a think and we can decide what to do here.